lab_2/main.py

```diff
@@ -89,8 +168,42 @@ def predict_language_knn(unknown_text_vector: list, known_text_vectors: list,
     :param k: the number of neighbors to choose label from
     :param metric: specific metric to use while calculating distance
     """
-    pass
-
+    if not isinstance(unknown_text_vector, list) or not \
+        isinstance(known_text_vectors, list) or not \
+            isinstance(language_labels, list) or not \
+            len(known_text_vectors) == len(language_labels):
+        return None
+    if metric == 'euclid':
+        vectors_distances = []
+        predicted_languages = []
+        languages_results = {}
+        for known_text_vector in known_text_vectors:
+            if not isinstance(known_text_vector, list):
+                return None
+            vectors_distances.append(calculate_distance(unknown_text_vector, known_text_vector))
+    elif metric == 'manhattan':
+        vectors_distances = []
+        predicted_languages = []
+        languages_results = {}
+        for known_text_vector in known_text_vectors:
+            if not isinstance(known_text_vector, list):
+                return None
+            vectors_distances.append(round(calculate_distance_manhattan(unknown_text_vector,
+                                                                        known_text_vector), 5))
+    else:
+        return None
+    sorted_vectors_distances = vectors_distances.copy()
+    sorted_vectors_distances.sort()
+    min_distance_vectors = sorted_vectors_distances[:k]
+    for min_distance_vector in min_distance_vectors:
```

Comment:
better to get index here

Reply:
Can you explain what you mean, please? I can't think of a way I could get index in `for` line or especially before of it